plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -108,6 +110,17 @@ public synchronized TrinoCatalog create(ConnectorIdentity identity)
             icebergCatalog = icebergCatalogInstance;
         }
 
-        return new TrinoRestCatalog(icebergCatalog, catalogName, sessionType, trinoVersion, typeManager, uniqueTableLocation);
+        ImmutableMap.Builder<String, String> credentials = ImmutableMap.builder();
+        Optional.ofNullable(securityProperties.get().get(TOKEN)).ifPresent(token -> credentials.put(TOKEN, token));
+        Optional.ofNullable(securityProperties.get().get(CREDENTIAL)).ifPresent(credential -> credentials.put(CREDENTIAL, credential));
+
+        return new TrinoRestCatalog(
```

Comment:
So, how did this work before? Seems like we've been putting the credentials in the wrong map?

Reply:
Yeah I think it never worked and it was not evident from Tabular as tabular's token expiration time is 24h vs Polaris 1h.